ts/utils/serve_decorator.py

```diff
@@ -0,0 +1,215 @@
+"""
+@serve decorator
+Original inspiration 
+https://github.com/aws/sagemaker-pytorch-inference-toolkit/blob/master/src/sagemaker_pytorch_serving_container/torchserve.py
+https://fastapi.tiangolo.com/
+"""
+from functools import wraps
+from typing import List, Callable, Union
+import os
+import logging 
+import inspect
+from ts.torch_handler.base_handler import BaseHandler
+
+
+logger = logging.getLogger()
+
+
+def serve(func):
+    """
+    @serve is a decorator meant to be used on top of an inference function
+    @serve
+    def inference():
+        perform_inference..
+
+    The goal of @serve is to simplify the handler authoring experience in torchserve
+
+    inference() can be as complex as you like and call other functions 
+
+    If you already have a handler you can manually call create_torchserve_config(), archive_model() 
+    and start_torchserve()
+
+    What this will do is
+    1. Override the base handler with a new handler with the inference function overloaded
+    2. Create a new torchserve config 
+    3. Archive a model assuming weights are in disk
+    4. Start torchserve
+
+    # TODO: Put everything in a temporary directory so user files don't get overwritten
+    """
+    @wraps(func)
+    def serve_model(*args, **kwargs):
+        """
+        Can override inference and management port or model name here
+        @serve(inference_http_port=8080, management_port=8081, serialized_file="my_model.pt", batch_size=32)
+        Full available configurations can be found in ts_config`
+
+        If you need to serve a full handler you can instead call create_handler, create_torchserve_config and torchserve_start manually
+        """
+        
+        handler = create_handler(func)
+
+        config = create_torchserve_config(inference_http_port=7070, management_http_port=7071, batch_size=32)
+        
+        model = archive_model(serialized_file="model.pt")
+
+        torchserve_start(handler= handler, models=model, config=config)
+
+        logger.warning("Don't forget to call torchserve --stop when you're done")
+        
+        return 0
+
+
+def create_handler(inference_func : Union[Callable, BaseHandler], handler_file : str = "handler.py"):
+    """
+    Create a new handler that inherits from the base handler
+    Update the inference() function with the function captured by the decorator
+    Write the handler to disk
+
+    TODO: Improve story for adding custom dependencies, all the hidden imports or library code
+    TODO: Improve workflow story, coming in future PR
+
+    """
+    
+    # If user provides a function then override inference() function in base handler
```

Comment:
@msaroufim is this general to any function? including inference, initialize, pre and post process?

Reply:
what is the reason that we only reference the inference func here?

Reply:
It's purely for simplicity, technically you only need the `inference()` function because you can do pre and post processing as well in the same function